test-data/unit/check-inference.test

```diff
@@ -791,10 +791,10 @@ AnyStr = TypeVar('AnyStr', bytes, str)
 def f(x: Union[AnyStr, int], *a: AnyStr) -> None: pass
 f('foo')
 f('foo', 'bar')
-f('foo', b'bar') # E: Value of type variable "AnyStr" of "f" cannot be "object"
+f('foo', b'bar') # E: Value of type variable "AnyStr" of "f" cannot be "Sequence[object]"
```

Comment:
This is due to `str` being a `Sequence[str]` and `bytes` being `Sequence[int]`, so they get joined to `Sequence[object]`.